### DISCUSSION TITLE
Suricata rule mismatch in 2.4.90

### DISCUSSION THREAD

**Discussion by Syngelik** [Security-Onion-Solutions/securityonion#13524] [2.4] [answered]
### Version

2.4.90

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

airgap

### Hardware Specs

Exceeds minimum requirements

### CPU

24

### RAM

64 GB

### Storage for /

500 GB

### Storage for /nsm

48 TB

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

My issue essentially mirrors the issue described in https://github.com/Security-Onion-Solutions/securityonion/discussions/13124

My initial installation was on release 2.4.60, after upgrading to 2.4.70 the issue began though instead of resolving as it did for a few in the above discussion after going to 2.4.80, I'm still experiencing it on 2.4.90.

![image](https://github.com/user-attachments/assets/59193f7c-1b76-4599-b633-2080aef4db34)
After clicking on suricata rule mismatch within the detections tab

![image](https://github.com/user-attachments/assets/876dd6fb-5592-472d-988c-d77258f09510)
The message found within each of the warn events shown on the first picture "integrity check failed"

I would say very few configuration changes in the grand scheme of things had been made prior to 2.4.80 and only a couple of tuning changes as well. Forcing a sync of any kind within detections changes nothing, there are no entries in local.rules, and the metadata engine is still Zeek. There are no errors when running salt-call state.highstate 

Please let me know what else I can try!

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by Carlos-mb** [comment]
Look at this: https://github.com/Security-Onion-Solutions/securityonion/discussions/13525

**Reply by Carlos-mb** [reply]
This solution is much better: https://github.com/Security-Onion-Solutions/securityonion/discussions/13238#discussioncomment-10408253

**Reply by Syngelik** [reply]
So I did come across this, but we're talking about a count of ~3500 total rules if I'm adding the two groups together. That's insane, is there no better way than manually enabling or disabling each rule?

**Reply by Carlos-mb** [reply]
I remember I was able to mark all rules listed using a filter and disable and enabled all together. Can you do that?

**Comment by defensivedepth** [comment]
@Syngelik Can you please post the output of the following, run from the Manager:

`sudo salt-call pillar.get idstools`

**Reply by Syngelik** [reply]
local:
    ----------
    enabled:
        True
    sids:
        ----------
        disabled:   
        enabled: 
        modify:
            - 2013031 "startswith;" "startswith; content: !\"redacted\"; nocase;"
            - 2013030 "startswith;" "startswith; content !\"redacted\"; nocase;"
            - 2013028 "startswith;" "startswith; content: !\"redacted\"; nocase;"

**Reply by Syngelik** [reply]
Is there a way to reset the rules that are enabled/disabled to default?

I'm thinking I can disable and enable every suricata rule in detections and that should "reset" their status... if that makes sense.

**Reply by Syngelik** [reply]
Well I tried doing the above for kicks because I didn't mind having all rules enabled. now there are just 17408 rules enabled but not deployed. All 53k and change ET Open rules are now showing under enabled from the above output now instead.

**Comment by defensivedepth** [comment]
@Syngelik Is there literally nothing in the `enabled` or `disabled` section when you run `sudo salt-call pillar.get idstools` ?

Also, please confirm what version you are on - .80 or .90 ?

**Reply by Syngelik** [reply]
I ended up disabling and enabling every single rule ET Open rule through the detections page to see if that would do anything. Now every rule is listed under enabled. There are now 0 rules listed as deployed but not enabled, but still thousands listed as enabled but not deployed.

I am on .90

**Reply by defensivedepth** [reply]
`Now every rule is listed under enabled. ` --> you mean when you run `sudo salt-call pillar.get idstools` ?

Please run `sudo so-rule-update` and post the last 10 lines of output.

**Reply by Syngelik** [reply]
>you mean when you run sudo salt-call pillar.get idstools

Correct, below is the output of so-rule-update. I run an airgap installation though technically have minimal internet access, very strict whitelist, so I'm not surprised so-rule-update doesn't work but the output is below anyway.

------------------------------------

Traceback (most recent call last):
  File "/usr/local/bin/idstools-rulecat", line 12, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 859, in main
    files = Fetch(args).run()
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 329, in run
    files.update(self.fetch(url))
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 317, in fetch
    idstools.net.get(
  File "/usr/local/lib/python3.9/site-packages/idstools/net.py", line 55, in get
    remote = urlopen(url)
  File "/usr/local/lib/python3.9/urllib/request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/local/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/local/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/local/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden

**Reply by defensivedepth** [reply]
eh, can you post the full output. ?

**Reply by Syngelik** [reply]
![image](https://github.com/user-attachments/assets/af5f4933-cf91-478f-9372-6b82eb60ac2b)

**Reply by Syngelik** [reply]
@defensivedepth Any other thoughts? Is there a way I can prevent it from trying to go out to the internet, presumably github?

**Reply by defensivedepth** [reply, marked as answer]
Can you doublecheck what version you are on?

**Reply by Syngelik** [reply]
![image](https://github.com/user-attachments/assets/ab2badbb-c915-4cc7-a8bd-d8f032585bc2)

**Reply by Syngelik** [reply]
Upgraded to .100, seems to have fixed it. Thanks.

**Reply by Syngelik** [reply]
@defensivedepth I lied, it blew up again. Kind of at a loss why the airgap installation insists on going out..?
![image](https://github.com/user-attachments/assets/660c1873-67ef-44ec-b4be-6c6188b23efa)

**Reply by Syngelik** [reply]
I removed my proxy from all configurations and that allowed me to run the rule-update command and fixed the issue.
